lib/engine/cube_chart.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module Engine
+  class CubeChart
+    attr_reader :header, :footer, :layout, :row_labels
```

Comment:
This is way too much view information and if it's anywhere, it belongs in the view.

The engine code should be defined to do what the engine needs to do without any knowledge of the view. It is the view code's job to take engine information and transform it into a visual representation.

Reply:
For example, par chart is used by the engine and contains the possible par locations for determining what the possible pars are and turn order in the game. All the code to represent that visually is in the view.

Reply:
Stock market is another good example